llvm/lib/SYCLLowerIR/LowerWGScope.cpp

```diff
@@ -779,13 +786,15 @@ PreservedAnalyses SYCLLowerWGScopePass::run(Function &F, const llvm::Triple &TT,
     // globals.
     Instruction *I = BB.getFirstNonPHI();
 
-    for (; I->getOpcode() == Instruction::Alloca; I = I->getNextNode()) {
+    for (; I->getOpcode() == Instruction::Alloca ||
```

Comment:
Maybe update the comment above since now you also skip addrspacecasts associated with allocas.

Reply:
Yes, thanks, fixed the comment.